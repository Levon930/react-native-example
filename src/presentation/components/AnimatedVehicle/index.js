import React, { useContext } from 'react';
import { Animated, Image, StyleSheet, View } from 'react-native';
import { Marker } from 'react-native-maps';
import { Overlay, Text } from '../';
import { LocalizationContext } from '../../../services';
import {getRouteAbbrev, getVehicleColor} from '../../../utils';

export default ({ item, showHandicapped }) => {
  const { appLanguage } = useContext(LocalizationContext);
  return (
    <If condition={(showHandicapped && item.handicapped) || (!showHandicapped)}>
      <Marker.Animated
        key={item.bort_number}
        ref={item.ref}
        coordinate={item.coordinate}
        deg={item.deg}
        tracksViewChanges={false}
        anchor={{ x: 0.5, y: 0.5 }}
        flat
        // style={{ // markers rotation in case if rotation will be enebled on the map
        //   transform: [{
        //     rotate: heading.interpolate({
        //       inputRange: [0, 360], outputRange: ['0deg', '360deg'],
        //     }),
        //   }],
        // }}
      >
        <Text bold style={styles.transportBortNumber}>
          {getRouteAbbrev(item.vehicleType, appLanguage)}{' '}{item.number}
        </Text>
        <Text style={styles.hidden}>{item.deg._value}</Text>
        <Image
          source={(item.icon)}
          style={styles.transportIcon}
        />
        <If condition={item.deg}>
          <Overlay style={styles.transportArrowWrapper}>
            <If condition={item.deg._value !== 0}>
              <Animated.View
                style={[{
                  transform: [{
                    rotate: item.deg.interpolate({
                      inputRange: [0, 360], outputRange: ['0deg', '360deg'],
                    }),
                  }],
                },
                styles.transportArrow]}
              >
                <View
                  style={[styles.directionIcon, {borderBottomColor: getVehicleColor(item.vehicleType)}]}
                />
              </Animated.View>
            </If>
          </Overlay>
        </If>
      </Marker.Animated>
    </If>
  );
};

const styles = StyleSheet.create({
  directionIcon: {
    width: 0,
    height: 0,
    backgroundColor: 'transparent',
    borderStyle: 'solid',
    borderLeftWidth: 5,
    borderRightWidth: 5,
    borderBottomWidth: 6,
    borderLeftColor: 'transparent',
    borderRightColor: 'transparent',
    // borderBottomColor: '#3D4763',
  },
  transportArrowWrapper: {
    paddingBottom: -100,
  },
  transportArrow: {
    alignItems: 'center',
    height: 40,
    width: 40,
    margin: 15,
  },
  transportBortNumber: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    textAlign: 'center',
  },
  transportIcon: {
    height: 50,
    width: 50,
    margin: 10,
  },
  hidden: {
    position: 'absolute',
    opacity: 0,
  },
});
