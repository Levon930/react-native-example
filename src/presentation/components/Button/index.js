import React from 'react';
import { TouchableOpacity, View } from 'react-native';
import { Text } from '../';
import styles from './styles';

function _getContainerStyle(props) {
  let containerStyle;
  if (props.filled) {
    containerStyle = styles.containerFilled;
  } else {
    containerStyle = styles.containerDefault;
  }
  return containerStyle;
}

export default ({ style, onPress, title, disabled, child, ...props }) => (
  <TouchableOpacity
    {...props}
    onPress={onPress}
    disabled={disabled}
    style={[disabled ? styles.containerDisabled : _getContainerStyle(props), style]}
  >
    <View>
      <If condition={child !== undefined}>
        {child}
      </If>
      <If condition={!child && title}>
        <Text style={props.filled ? [styles.text, styles.textReverted] : styles.text}>{title}</Text>
      </If>
    </View>
  </TouchableOpacity>
);
