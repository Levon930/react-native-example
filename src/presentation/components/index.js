export { SafeAreaLayout } from './SafeAreaLayout';
export { default as TabBar } from './TabBar';
export { default as Overlay } from './Overlay';
export { default as Row } from './Row';
export { SearchTopPanel } from './SearchTopPanel';
export { FloatingTitleTextInputField } from './FloatingTitleTextInputField';
export { default as Text } from './Text';
export { default as SearchInput } from './SearchInput';
export { default as Button } from './Button';
export { default as TextInput } from './TextInput';
export { default as Header } from './Header';
export { default as Divider } from './Divider';
export { default as StopItem } from './StopItem';
export { default as StickyHeader } from './StickyHeader';
export { default as TogglingStops } from './TogglingStops';
export { default as ClickableIcon } from './ClickableIcon';
export { default as Map } from './Map';
export { default as VerticalDots } from './VerticalDots';
export { default as ETAModal } from './ETAModal';
export { default as VehicleIcon } from './VehicleIcon';
export { default as RouteListItem } from './RouteListItem';
export { default as EndlessScrollView } from './EndlessScrollView';
export { default as LoaderView } from './LoaderView';
export { default as AnimatedVehicle } from './AnimatedVehicle';
