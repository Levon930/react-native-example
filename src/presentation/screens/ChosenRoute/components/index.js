export { TitledText } from './TitledText';
export { ClickableDirectionTab } from './ClickableDirectionTab';
