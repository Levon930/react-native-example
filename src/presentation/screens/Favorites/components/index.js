export { default as TabItem } from './TabItem';
export { default as StopsListItem } from './StopsListItem';
