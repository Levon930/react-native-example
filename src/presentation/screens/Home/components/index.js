export { default as ClickableListItem } from './ClickableListItem';
export { default as FadeInView } from './FadeInView';
export { default as RouteList } from './RouteList';
export { default as FavoriteStop } from './FavoriteStop';
export { default as FilterButton } from './FilterButton';
