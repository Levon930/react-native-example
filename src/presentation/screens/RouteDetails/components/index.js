export { TitledText } from './TitledText';
export { default as ListItem } from './ListItem';
