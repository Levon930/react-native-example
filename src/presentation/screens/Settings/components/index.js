export { default as ItemSection } from './ItemSection';
export { default as SectionHeader } from './SectionHeader';
