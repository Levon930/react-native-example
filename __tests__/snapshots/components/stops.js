
import React from 'react';
import renderer from 'react-test-renderer';
import { TogglingStops, StopItem } from 'components';

const stop = {
  index: 0,
  id: '5e7f2e7115ad5b4c0414426c',
  isStart: true,
  nameEN: 'Залізничний вокзал',
  nameRU: 'Залізничний вокзал',
  nameUA: 'Залізничний вокзал',
  number: 48207,
  geolocation: { latitude: 49.835381, longitude: 23.995241 },
};

const mockTrackedRouteData = [
  {
    gpsStatus: true,
    id: '5e7f2e8015ad5b4c0414428b',
    isSuburban: false,
    length: 0,
    costMoving: 7,
    startHour: '2020-03-28T11:01:20.561Z',
    endHour: '2020-03-28T12:01:20.561Z',
    number: '1',
    tempRoute: false,
    type: 0,
    forward: {
      pathId: '5e7f2e8015ad5b4c04144287',
      geolocation: [
        { id: 0, latitude: 48.475866446966, longitude: 35.015351474285 },
      ],
      stops: [
        stop,
      ],
    },
    backward: {
      pathId: '4_25634758',
      geolocation: [
        { id: 1011, latitude: 49.835443, longitude: 23.995359 },
      ],
      stops: [
        {
          index: 0,
          id: '5e7f2e7915ad5b4c0414427b',
          isStart: true,
          nameEN: 'Транспортний університет',
          nameRU: 'Транспортний університет',
          nameUA: 'Транспортний університет',
          number: 48207,
          geolocation: { latitude: 48.438530429539, longitude: 35.05035007 },
        },
      ],
    },
  },
];

test('TogglingStops render correctly', () => {

  const renderStopView = () => {};
  const onStopPress = () => {};
  const renderStartStopFlag = () => null;
  const renderEndStopFlag = () => null;

  const tree = renderer.create(
    <TogglingStops
      routesDataSet={mockTrackedRouteData}
      showSelectRouteOnly
      zoomThreadhold={15}
      renderStartStopFlag={renderStartStopFlag}
      regionToReactTo={{}}
      routeIdToReactTo={''}
      renderStopView={renderStopView}
      renderEndStopFlag={renderEndStopFlag}
      onStopPress={onStopPress}
    />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Stop general component render correctly', () => {

  const tree = renderer.create(
    <StopItem
      item={stop}
      index={stop.index}
      onPress={() => {}}
    />).toJSON();
  expect(tree).toMatchSnapshot();
});
