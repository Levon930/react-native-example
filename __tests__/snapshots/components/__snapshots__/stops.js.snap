// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stop general component render correctly 1`] = `
<Marker
  coordinate={
    Object {
      "latitude": 49.835381,
      "longitude": 23.995241,
    }
  }
  onPress={[Function]}
  tracksViewChanges={false}
>
  <View
    style={
      Object {
        "padding": 4,
      }
    }
  >
    <Image />
  </View>
</Marker>
`;

exports[`TogglingStops render correctly 1`] = `
Array [
  <If
    condition={false}
  >
    <View>
      <If
        condition={true}
      >
        <Marker
          coordinate={
            Object {
              "latitude": 49.835381,
              "longitude": 23.995241,
            }
          }
          onPress={[Function]}
          tracksViewChanges={false}
        >
          <View />
        </Marker>
      </If>
      <If>
        <Marker
          coordinate={
            Object {
              "latitude": 49.835381,
              "longitude": 23.995241,
            }
          }
          onPress={[Function]}
          tracksViewChanges={false}
        >
          <View />
        </Marker>
      </If>
    </View>
  </If>,
  <If
    condition={false}
  >
    <View>
      <If
        condition={true}
      >
        <Marker
          coordinate={
            Object {
              "latitude": 48.438530429539,
              "longitude": 35.05035007,
            }
          }
          onPress={[Function]}
          tracksViewChanges={false}
        >
          <View />
        </Marker>
      </If>
      <If>
        <Marker
          coordinate={
            Object {
              "latitude": 48.438530429539,
              "longitude": 35.05035007,
            }
          }
          onPress={[Function]}
          tracksViewChanges={false}
        >
          <View />
        </Marker>
      </If>
    </View>
  </If>,
]
`;
